src/Interpreters/executeQuery.cpp

```diff
@@ -626,6 +631,27 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
         }
         else
         {
```

Comment:
GitHub does not allow to add a comment for the `if` branch above (because it's unchanged), but it makes sense to throw `NOT_IMPLEMENTED` if both `async_insert` and `implicit_transaction` are `true`.

Reply:
I guess that this should be done only if `throw_on_unsupported_query_inside_transaction` is on, right?